lib/pbench/cli/agent/commands/results/push.py

```diff
@@ -0,0 +1,66 @@
+import click
+
+from pbench.agent.base import BaseCommand
+from pbench.agent.results import CopyResultTb
+from pbench.cli.agent import pass_cli_context, CliContext
+from pbench.cli.agent.options import common_options
+from pbench.common.utils import md5sum
+
+
+class ResultsPush(BaseCommand):
+    def __init__(self, context: CliContext):
+        super().__init__(context)
+
+    def execute(self) -> int:
+        tarball_len, tarball_md5 = md5sum(self.context.result_tb_name)
+        crt = CopyResultTb(
+            self.context.controller,
+            self.context.result_tb_name,
+            tarball_len,
+            tarball_md5,
+            self.config,
+            self.logger,
+        )
+        crt.copy_result_tb(self.context.token)
+        return 0
+
+
+@click.command("pbench-results-push")
+@common_options
+@click.option(
+    "--token",
+    required=True,
+    prompt=True,
+    envvar="PBENCH_ACCESS_TOKEN",
+    help="pbench server authentication token (will prompt if unspecified)",
+)
+@click.argument("controller")
+@click.argument(
+    "result_tb_name",
+    type=click.Path(exists=True, file_okay=True, dir_okay=False, readable=True),
+)
+@pass_cli_context
+def main(
+    context: CliContext, controller: str, result_tb_name: str, token: str,
+):
+    """Push a result tar ball to the configured Pbench server.
+
+        \b
+        CONTROLLER is the name of the controlling node.
+        RESULT_TB_NAME is the path to the result tar ball.
+        \f
+        (This docstring will be printed as the help text for this command;
+        the backslash-escaped letters are formatting directives; this
+        parenthetical text will not appear in the help output.)
+    """
+    context.controller = controller
+    context.result_tb_name = result_tb_name
+    context.token = token
+
+    try:
+        rv = ResultsPush(context).execute()
+    except Exception as exc:
+        click.echo(exc, err=True)
+        rv = 1
```

Comment:
I concur...but, I'm not sure it's worth it:  we would have to figure out which exceptions should map to which error code, and then add all the appropriate conditionals, and I think it would yield very limited value.  And, I think the case that you cited is already covered (although, it also returns `1` instead of `2`, which is undesirable, but it gets dealt with before `main()` gets called.)